### PR TITLE
OCPCLOUD-2058: Update k8s to 1.27

### PR DESCRIPTION
**Pull request by RadekManak**
Updates kubernetes dependencies to 1.27
/hold for machine-api-operator k8s bump merge

### CODE REVIEW

**Comment by odvarkadaniel** [comment]
I will close my PR #60 that solves this issue https://issues.redhat.com/browse/OCPBUGS-5723 as it is a subset of what needs to be done here.

File: cmd/manager/main.go [hunk 1]
```diff
@@ -215,12 +214,15 @@ func newConfigManagedClient(mgr manager.Manager) (runtimeclient.Client, manager.
 	clientOpts := runtimeclient.Options{
 		Scheme: mgr.GetScheme(),
 		Mapper: mgr.GetRESTMapper(),
+		Cache: &runtimeclient.CacheOptions{
+			Reader: cache,
+		},
 	}
 
-	cachedClient, err := cluster.DefaultNewClient(cache, config.GetConfigOrDie(), clientOpts)
+	clinet, err := runtimeclient.New(config.GetConfigOrDie(), clientOpts)
```
Comment (JoelSpeed) [inline cmd/manager/main.go:222]: What's a `clinet`? 😉

**Pull request merged**